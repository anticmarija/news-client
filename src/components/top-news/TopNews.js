import React, { useEffect } from "react";
import { withRouter } from "react-router-dom";
import { useSelector, useDispatch } from "react-redux";
import { fetchArticles } from "../../store/actions/top-news.actions";
import Thumbnail from "../thumbnail/Thumbnail";
import Loader from "../loader/Loader";
import { MainHeaderStyle, NewsWrapperStyle } from "../../style/Shared.style";
import { useTranslation } from "react-i18next";

const TopNews = () => {
  const { t } = useTranslation();

  const { articles, loading } = useSelector((state) => state.topNews);
  const dispatch = useDispatch();

  const { activeCountry } = useSelector((state) => state.countries);

  useEffect(() => {
    dispatch(fetchArticles(activeCountry));
    // eslint-disable-next-line
  }, [activeCountry]);

  return (
    <div>
      <MainHeaderStyle>
        {t("topNews.header")} {t(`countries.${activeCountry}`)}
      </MainHeaderStyle>
      {loading ? (
        <Loader />
      ) : (
        <NewsWrapperStyle>
          {articles?.map((article) => (
            <Thumbnail
              key={`${article.title}_${article.publishedAt}`}
              article={article}
            />
          ))}
        </NewsWrapperStyle>
      )}
    </div>
  );
};

export default withRouter(TopNews);
