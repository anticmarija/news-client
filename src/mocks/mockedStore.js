import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];
export const initialState = {
  topNews: {},
  countries: {
    supportedCountries: [{ countryId: "gb", countryName: "Gb" }],
  },
  categories: {},
  article: {
    activeArticle: {
      title: "active",
      urlToImage:
        "https://i.dailymail.co.uk/1s/2020/06/17/16/29719992-0-image-a-23_1592408042536.jpg",
      description: "desc",
    },
  },
};

export const mockedStore = (state = initialState) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state);
};
