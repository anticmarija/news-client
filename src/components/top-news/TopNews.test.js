import React from "react";
import TopNews from "./TopNews";
import { mockedStore } from "../../mocks/mockedStore";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

const store = mockedStore();

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (s) => s }),
}));

describe("Test TopNews component", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <TopNews />
        </MemoryRouter>
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
