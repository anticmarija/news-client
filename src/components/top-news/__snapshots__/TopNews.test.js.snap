// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TopNews component should render correctly 1`] = `
<div>
  <h1
    class="Sharedstyle__MainHeaderStyle-sc-1a3bnp9-0 chhIjm"
  >
    topNews.header
     
    countries.gb
  </h1>
  <div
    class="Sharedstyle__NewsWrapperStyle-sc-1a3bnp9-1 jTferJ"
  >
    <div
      class="Thumbnailstyle__ThumbnailWrapperStyle-sc-123obdz-0 lhHvSI"
    >
      <h2
        class="Thumbnailstyle__ThumbnailTitleStyle-sc-123obdz-3 dSnuxn"
      >
        article
      </h2>
      <div
        class="Thumbnailstyle__ThumbnailImgStyle-sc-123obdz-1 eTaMhC"
      />
      <div
        class="Thumbnailstyle__ThumbnailDescStyle-sc-123obdz-2 dPDhAr"
      >
        <div>
          desc
        </div>
        <span
          class="Thumbnailstyle__ThumbnailLinkStyle-sc-123obdz-4 VvDHO"
        >
          <a
            data-testid="more-link"
            href="/country/gb/articles/article"
          >
            thumbnail.readMore
             &gt;
          </a>
        </span>
      </div>
    </div>
  </div>
</div>
`;
